backend/controller/SequenceController.ts

```
@@ -0,0 +1,384 @@
+                user.sharedSequencesReadAccess,
+                code
+            );
+            user.save();
+        }
+        for (const x of seq.writeAccess) {
+            const user = await DBUser.findOneBy({ id: x });
+            if (user == null) {
+                continue;
+            }
+            user.sharedSequencesWriteAccess = this.removeFromUserList(
+                user.sharedSequencesWriteAccess,
+                code
+            );
+            user.save();
+        }
+        seq.remove();
+    }
+
+    /**
+     * Searches all sequenes of the user
+     * @param userId the id of the user
+     * @returns the sequences of the user
+     */
+    public static async getSequencesOfUser(
+        userId: number
+    ): Promise<ISequence[]> {
+        if (!(await this.isValidUser(userId))) {
+            throw new LoernwerkError(
+                'User doesnt exists',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        return await DBSequence.find({ where: { authorId: userId } });
+    }
+
+    /**
+     * Searches all sequences shared with the user.
+     * @param userId the id of the user
+     * @returns the sequences shared with the user
+     */
+    public static async getSharedSequencesOfUser(
+        userId: number
+    ): Promise<ISequence[]> {
+        const user = await DBUser.findOneBy({ id: userId });
+        if (user === null) {
+            throw new LoernwerkError(
+                'User doesnt exists',
+                LoernwerkErrorCodes.NOT_FOUND
+            );
+        }
+        const seqlist: ISequence[] = [];
+        for (const x of user.sharedSequencesReadAccess.concat(
+            user.sharedSequencesWriteAccess
+        )) {
+            const seq = await DBSequence.findOneBy({ code: x });
+            if (seq === null) {
+                // May just bring it back to consistency?
+                throw new LoernwerkError(
```

Should we throw an error, or just log it to console, but keep going?

---

Could be a problem if we want to add someone to an access and it fails, then an error would be good.

---

fair point, shouldn't happen either way